errortypes/scope.go

```diff
@@ -0,0 +1,19 @@
+package errortypes
+
+type Scope int
+
+const (
+	ScopeAny Scope = iota
+	ScopeDebug
+)
+
+type Scoped interface {
+	Scope() Scope
+}
+
+func ReadScope(err error) Scope {
```

Comment:
Is it possible to add a test for this function?

Reply:
done